### PR TITLE
[DRAFT] chore(storage-browser): replace history with location

### PR DESCRIPTION
**Pull request by calebpollman**
#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] `yarn test` passes and tests are updated/added
- [ ] PR title and commit messages follow [conventional commit](https://www.conventionalcommits.org/en/v1.0.0/#summary) syntax
- [ ] _If this change should result in a version bump_, [changeset added](https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md) (This can be done after creating the PR.) This does not apply to changes made to `docs`, `e2e`, `examples`, or other private packages.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-storage/src/components/StorageBrowser/providers/store/location/context.tsx [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+export interface LocationState {
+  /**
+   * current `location` metadata
+   */
+  location: LocationData | undefined;
+
+  /**
+   * current `location` subpath within a `location`
+   */
+  path: string | undefined;
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/providers/store/location/context.tsx:49]: `path` updates on `location` change events with a `location`:
```
// <bucket>/<prefix>/<path>
// initial: 'my-bucket/my-prefix/'
const path = undefined;

// enter nested location: 'my-bucket/my-prefix/a-folder/'
const path = 'a-folder/';

// enter deeply nested folder: 'my-bucket/my-prefix/a-folder/extra-nested/further-nexted/'
const path = 'a-folder/extra-nested/further-nested/';

// navigate up two levels: 'my-bucket/my-prefix/a-folder/'
const path = 'a-folder/;
```

File: packages/react-storage/src/components/StorageBrowser/providers/store/location/context.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+ *   },
+ *   path: 'a-path/a-nested-path/',
+ * }
+ *
+ */
+export interface LocationState {
+  /**
+   * current `location` metadata
+   */
+  location: LocationData | undefined;
```
Comment (calebpollman) [inline packages/react-storage/src/components/StorageBrowser/providers/store/location/context.tsx:44]: `location` stays stable between navigation events within a `location`, e.g. entering/exiting a nested folder

File: packages/react-storage/src/components/StorageBrowser/providers/store/location/index.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+export {
+  HistoryActionType,
+  HistoryProvider,
+  HistoryState,
+  HistoryProviderProps,
+  useHistory,
+} from './context';
```
Comment (cshfang) [inline packages/react-storage/src/components/StorageBrowser/providers/store/location/index.ts:7]: History?

Comment (calebpollman) [reply packages/react-storage/src/components/StorageBrowser/providers/store/location/index.ts:7]: "Cmd + c" -> "Cmd + v" problems

**Pull request closed without merging**